File: src/main/java/corp/soyummy/recipes/RecipeRepository.java
package corp.soyummy.recipes;

import org.springframework.stereotype.Repository;
import org.springframework.data.mongodb.repository.MongoRepository;

import java.util.List;

@Repository
public interface RecipeRepository extends MongoRepository<Recipe, String> {
    List<Recipe> findByTitleContainingIgnoreCase(String title);
    List<Recipe> findByCategoryContainingIgnoreCase(String category);
    List<Recipe> findByAreaContainingIgnoreCase(String area);
    List<Recipe> findByTagsContainingIgnoreCase(String tag);
}

File: src/main/kotlin/com/soyummy/recipes/RecipeRepository.kt
package com.soyummy.recipes

import org.springframework.stereotype.Repository
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

@Repository
interface RecipeRepository : MongoRepository<Recipe, String> {
    fun findByTitleContainingIgnoreCase(title: String, pageable: Pageable): Page<Recipe>
    fun findByCategoryContainingIgnoreCase(category: String, pageable: Pageable): Page<Recipe>
    fun findByAreaContainingIgnoreCase(area: String, pageable: Pageable): Page<Recipe>
    fun findByTagsContainingIgnoreCase(tag: String, pageable: Pageable): Page<Recipe>
}